### PR TITLE
Stack Recipe Lists

### PR DESCRIPTION
**Pull request by uraniummeltdown**
srl

Coming soon: manual airlock assembly construction that doesn't suck

:cl:
add: Stack recipes now have support for lists
tweak: Stack menus use /datum/browser
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -87,57 +88,76 @@
...
-					t1 += " <A href='?src=[REF(src)];make=[i];multiplier=[n]'>[n*R.res_amount]x</A>"
-			if (!(max_multiplier in multipliers))
-				t1 += " <A href='?src=[REF(src)];make=[i];multiplier=[max_multiplier]'>[max_multiplier*R.res_amount]x</A>"
-
-	t1 += "</TT></body></HTML>"
-	user << browse(t1, "window=stack")
+
+		if (istype(E, /datum/stack_recipe_list))
+			var/datum/stack_recipe_list/srl = E
+			t1 += "<a href='?src=\ref[src];sublist=[i]'>[srl.title]</a>"
```
Comment (SamHPurp) [inline code/game/objects/items/stacks/stack.dm]: We can't use `\ref[src]` anymore due to 512

Use `[REF(src)]`

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -87,57 +88,76 @@
...
+			var/can_build = 1
+			can_build = can_build && (max_multiplier>0)
+
+			if (R.res_amount>1)
+				title+= "[R.res_amount]x [R.title]\s"
+			else
+				title+= "[R.title]"
+			title+= " ([R.req_amount] [singular_name]\s)"
+			if (can_build)
+				t1 += text("<A href='?src=\ref[src];sublist=[recipes_sublist];make=[i];multiplier=1'>[title]</A>  ")
```
Comment (SamHPurp) [inline code/game/objects/items/stacks/stack.dm]: same here

File: code/game/objects/items/stacks/stack.dm [hunk 3]
```diff
@@ -87,57 +88,76 @@
...
+			else
+				t1 += text("[]", title)
+				continue
+			if (R.max_res_amount>1 && max_multiplier>1)
+				max_multiplier = min(max_multiplier, round(R.max_res_amount/R.res_amount))
+				t1 += " |"
+				var/list/multipliers = list(5,10,25)
+				for (var/n in multipliers)
+					if (max_multiplier>=n)
+						t1 += " <A href='?src=\ref[src];make=[i];multiplier=[n]'>[n*R.res_amount]x</A>"
```
Comment (SamHPurp) [inline code/game/objects/items/stacks/stack.dm]: and here

File: code/game/objects/items/stacks/stack.dm [hunk 4]
```diff
@@ -87,57 +88,76 @@
...
+				if (!(max_multiplier in multipliers))
+					t1 += " <A href='?src=\ref[src];make=[i];multiplier=[max_multiplier]'>[max_multiplier*R.res_amount]x</A>"
```
Comment (SamHPurp) [inline code/game/objects/items/stacks/stack.dm]: and here

**Comment by Xantholne** [comment]
QOL port.

**Pull request merged**